core/src/main/java/bisq/core/trade/TradeManager.java

```diff
@@ -269,10 +271,15 @@ private void handleTakeOfferRequest(NodeAddress peer, InputsForDepositTxRequest
                     openOffer.getMediatorNodeAddress(),
                     openOffer.getRefundAgentNodeAddress(),
                     btcWalletService,
-                    getNewProcessModel(offer));
+                    getNewProcessModel(offer),
+                    UUID.randomUUID().toString());
         }
         TradeProtocol tradeProtocol = TradeProtocolFactory.getNewTradeProtocol(trade);
-        tradeProtocolByTradeId.put(trade.getId(), tradeProtocol);
+        TradeProtocol prev = tradeProtocolByTradeId.put(trade.getUid(), tradeProtocol);
+        if (prev != null) {
+            log.error("We had already an entry with uid {}", trade.getUid());
```

Comment:
How would this happen? Collision of randomUUID?

Reply:
If for some reason the same trade is entering those methods. Should not be possible but prefer to keep the log to detect it if there is still some bug.